integration-cli/docker_api_logs_test.go

```diff
@@ -27,7 +27,14 @@ func (s *DockerSuite) TestLogsApiWithStdout(c *check.C) {
 
 	go func() {
 		res, body, err := sockRequestRaw("GET", fmt.Sprintf("/containers/%s/logs?follow=1&stdout=1&timestamps=1", id), nil, "")
-		out, _ := bufio.NewReader(body).ReadString('\n')
+		if err != nil {
+			chLog <- logOut{"", nil, err}
+		}
+		out, err := bufio.NewReader(body).ReadString('\n')
+		if err != nil {
+			chLog <- logOut{"", nil, err}
+		}
+		body.Close()
```

Comment:
should this be in a defer and moved to before line 33?  That way if body isn't nil but something goes wrong with the ReadString() we'll still do body.Close()?

Also, should there be a return after each chLog <- ...  in those error cases?